[11.0] mig base geoengine

asaunier opened the pull request:
Replaces initial PR at https://github.com/OCA/geospatial/pull/176
Requires changes in module base_geoengine_demo as in https://github.com/OCA/geospatial/pull/177

* [x] replay commits with git am
* [ ]  squash commits => at least the new commits are squashed
* [ ]  migrate base_geoengine
  * [x] py2 to py3
  * [x] make the geoengine view work
  * [x] make the main widget work (geo_edit_map)
  * [x] make the other geo widgets work
  * [ ] use inheritance of base model to create a mixin
* [ ] update OL version


Know issues:

* Multi selection filter is not displayed in the search view (need to find how to create it)
* A widget map in a `<notebook><page>` in a modal view won't be displayed, it is shown only if you switch to an other tab and come back on the first tab. An issue with event trigger is the cause.
* Minified version of openlayers is never used, it should when not in debug mode.
* WMTS has issue with reprojections.
* Currently updated to last version of OL3, OL4 and OL5 have some breaking changes we didn't dived into.
* Some travis issues (psycopg, js lint).

Commit:
[FIX] upgrade OpenLayers to 2.12

Commit:
[FIX] OpenStreetMap spelling

Commit:
[ADD] new MapBox raster layer type

Commit:
[FIX] license in base_geoengine

Commit:
[FIX] javascript semicolons

yvaucher committed:
[ADD] base_geoengine - text widget geo_point_xy

Commit:
[FIX] minor style changes

yvaucher committed:
[ADD] base_geoengine - text widget rectangle for polygones

Commit:
[FIX] fix the validate, when a field was empty, it was receiving a 0 instead of a false value, due to the min/max functions

yvaucher committed:
[FIX] base_geoengine - disable validate function for geo_point widget in view mode

yvaucher committed:
[FIX] base_geoengine - disable validate function for geo_rect widget in view mode

Commit:
[FIX] xy and bbox widget: correctly set the inputs as readonly when the attrs is readonly, some cleanings

nbessi committed:
[FIX] dependencies + add edit map widget requierments

nbessi committed:
[ADD] function to get edit map widget informations

nbessi committed:
[ADD] use to edit background property on raster layers

nbessi committed:
[IMP] add CSS and XML for Edit map widget

nbessi committed:
[MV] rename file

nbessi committed:
[ADD] support of validate and readonly on geoengine edit view

nbessi committed:
[ADD] invisible support

Commit:
update openlayers css style

Commit:
coding style

Commit:
coding style

Commit:
coding style, readonly FieldGeoEnginEditMap

Commit:
geometry creation

Commit:
swisstopo layer, projection support

Commit:
remove unused proj4js-combined.js

Commit:
edit points

nbessi committed:
[FIX] context sensitive require in raster layer views

nbessi committed:
[FIX] Safari compatibility of edit widget

nbessi committed:
[IMP] Typo and cleanup

nbessi committed:
[FIX] srid data sent to js

nbessi committed:
[FIX] debug data in xml

nbessi committed:
[FIX] call to super in geo fields

nbessi committed:
[TYPO]

nbessi committed:
[FIX] rm debug statement

yvaucher committed:
[IMP] base_geoengine - replace geoengine icons by better icons for Luc

nbessi committed:
[FIX] Readonly widget

yvaucher committed:
[FIX] base_geoengine - prevent the call of other widget get_value inside buildcontext in geo_edit_map widget form view. To make sure it isn't called before start method have been called for those widget

yvaucher committed:
[FIX] base_geoengine - use https for swisstopo layers to avoid blank maps from a https environement

nbessi committed:
[INIT] V7.0 serie

Commit:
Fixed warnings and view. Still has js errors.

Commit:
Javascript modified for 6.1 -> 7.0
openerp.web.form.Field -> openerp.web.form.AbstractField
openerp.web.page.readonly -> openerp.web.form.widget

Commit:
widget.type is now widget.widget

Commit:
Replaced Icon with 7.0 font for switch and removed depricated images

Commit:
Now displays map.
widget.element_id -> widget.id_for_label
$element -> $el

Commit:
Display geometry and zoom

Commit:
[FIX] Edit and create mode

Commit:
[FIX] Setting geometry values to database
set_value_from_ui -> on_ui_change

Commit:
[FIX] Demo Retail Machine
geoengine_raster_layer: mapbox_type -> mapbox_id
Merged some code from 6.1 branch

Commit:
[IMP] Colspans of Backend forms

Commit:
[IMP] Use widget name instead of id_for_label

nbessi committed:
[FIX] installable state of addons to be uninstallable. They do not match actual requirement to be installable

Commit:
Revert "Rename all addons to xxx_unported to be travis compliant"

This reverts commit 4a89ecbce3802028f69676b7d4b925d153687d80.

nbessi committed:
Fix non protected import of python dependences

nbessi committed:
Fix import protection in geo helpers

lmignon committed:
Make modules intallable

lmignon committed:
[MIG] geospatial widgets

lmignon committed:
forward port change from 6.1

lmignon committed:
[FIX] correctly support readonly attribute on fields.geo_point

lmignon committed:
Put only two numbers for version.

lmignon committed:
[PEP8] base_geoengine

lmignon committed:
[FIX] postgis extension initialization

lmignon committed:
[PEP8] base_geoengine_demo

lmignon committed:
[PYLINT] base_geoengine

lmignon committed:
[IMP] Add basic tests

lmignon committed:
[FIX] move all addons from __unported__

lmignon committed:
[FIX] Uses the new API

lmignon committed:
[IMP] Use the non minified version of Openelayers.js
Js are minified by Odoo in normal mode but keep as initial in debug mode

lmignon committed:
[FIX] Defines Fields for our specialized columns

lmignon committed:
[FIX] Follows V8 conventions
remove deprecated entries
include js dependencies in a override of web.assets template

lmignon committed:
[FIX] View inheritance

lmignon committed:
[FIX] Properly displays the map
* Renders the map only when its element is displayed since the size must be known to avoid errors
* Remove unused pager
* Fix declared style for svg elements. The size must be known and fixed for svg elements of map.

lmignon committed:
[FIX] Fix issues when google is used as Raster Layer Type
* only render map when the anchor element is visible
* load google JS in web.assets_common

lmignon committed:
[FIX] Use an init hook to initialize postgis extension at install

lmignon committed:
[FIX] An other fix when google is used as Raster Layer Type
The call to animate on the div element before the map is rendered introduces an offset between the google map layer and others layers

lmignon committed:
[IMP] Extract description in __openerp__.py to README.rst

lmignon committed:
[FIX] Get the tab id from the closest div with class='oe_notebook_page' element in place of the parent

lmignon committed:
[FIX] remove unused code

lmignon committed:
[FIX] move the group definitions into the security folder

lmignon committed:
[FIX] list all know contributors from git

lmignon committed:
[IMP]Add documentation for new fields

lmignon committed:
[IMP]Add documentation for the init_hook

lmignon committed:
[FIX] Reraise error

lmignon committed:
[FIX] Use a more precise type error

lmignon committed:
[IMP] Use triple quotes for sql query

lmignon committed:
[FIX] Reworks docstring

lmignon committed:
[FIX] Use a more precise type error

lmignon committed:
[FIX] + is redundant

lmignon committed:
[IMP] Add documentation

lmignon committed:
[FIX] Use a more precise exception

lmignon committed:
[FIX] Reworks docstring

lmignon committed:
[IMP] Use V8 API

lmignon committed:
[FIX] Use isinstance operator in place of the previous hack
Shapely now provide a BaseGeometry class as base class for all Geometry objects

Commit:
[FIX] run tests in V8

Commit:
[FIX] With new API: Non stored fields doesn't have column attribute

lmignon committed:
[ADD] Add acsone in the list of authors

nbessi committed:
Fix MissingError wrong argument number

And improve error message

nbessi committed:
Improve error message

nbessi committed:
Styling

nbessi committed:
Fix styling

nbessi committed:
Style

nbessi committed:
Experiment to remove of history

nbessi committed:
Add type testing when browsing

nbessi committed:
Avoid exception when query result is empty on map view

nbessi committed:
White space cleanup

nbessi committed:
Remove experimental dead code

Commit:
Allows doc to be build on github pages

lmignon committed:
[FIX] tests

lmignon committed:
[FIX] Declare admin user as member of 'Geoengine admin' group

Commit:
Fix travis documentation location

Commit:
Upgrade doc to postgis 2.0

lmignon committed:
[FIX] Add bug tracker url to the module description

Commit:
Fix addons path by loading them and pep8

lmignon committed:
Use the Nominatim service of OpenStreetMap to retrieve OSM data by address

lmignon committed:
[PEP8]

Commit:
Fix autodoc

Commit:
Fix docstring indentation

Commit:
Add pointer to documentation

Commit:
Fix copyright to OCA

Commit:
Renaming to geospatial

lmignon committed:
[IMP] OL3

lasley committed:
* updated Google maps URI to relative scheme
* updated js.mapbox theme URI to relative scheme and their Cloudfront address so that SSL cert is valid

lmignon committed:
[FIX] map styling and fullscreen mode

lmignon committed:
chromojs

lmignon committed:
FIX legend
Implement propotional
Improve alpha channel management (everythong is ready to enable its configuration on the raster layer)

lmignon committed:
[FIX] When testing geoegine view, only keep relevant keys from values returned by fields_view_get

dreispt committed:
Update installation instructions

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

yvaucher committed:
[MIGR] fix menu icons path for base_geoengine

yvaucher committed:
Remove version="7.0" from tests

yvaucher committed:
[MIGR] base_geoengine
- separate view and widgets in 2 files
- change js models using 'odoo.define'

yvaucher committed:
[MIGR] Separate common function for widget and view to create layers in a mixin abstract model

yvaucher committed:
[MIGR][FIX] Don't call order_by if order is not defined

yvaucher committed:
[MIGR] ensure geoengine view is loaded after container is rendered

yvaucher committed:
Not pretending we implement limit and offset

yvaucher committed:
Fix search returning 0 elements on geo view

yvaucher committed:
Patch mapfish to avoid trying to make more classes than element to classify

mdietrichc2c committed:
Improve patch to classify points to make sure there are as many classes as possible

yvaucher committed:
Fix a typo on default key

yvaucher committed:
Fix tab event listener to display map widget in form view in last tabs

yvaucher committed:
[FIX] display translation of field string instead of field name and display values of selection field instead of technical value in info map box in geoengine view

yvaucher committed:
[FIX] base_geoengine - allow to use multiple icons on your maps

The feature to set a symbol with symbol_url or symbol_binary on vector
layers was broken. This replace those fields by a new object vector
symbol on which you can specify which symbol to use depending on a field
value.

yvaucher committed:
Allow to drag pan through polygons

yvaucher committed:
Refactor WMTS to be able to add a generic config

And extract Swisstopo to move it in a new module

yvaucher committed:
Fix srid value read which was unset in field properties

yvaucher committed:
By default use OSM

yvaucher committed:
Add parameters to define extends and projection

yvaucher committed:
Fix path of openerp for doc build

yvaucher committed:
Remove google from selection as it is currently broken

yvaucher committed:
Add geo_contains operator to geo_search method.

geo_contains calls St_Contains SQL search.

Plus fix search to set SRID to avoid mixin srid issue.

yvaucher committed:
Target only the right tab where the map is to add listener.

fredj committed:
Ensure we call add_tab_listener only once

yvaucher committed:
Add a button on map widget to clear geometry from the map

yvaucher committed:
Fix _setup_fields signature due to OCA/OCB@87d160a
Otherwise partial value is interpreted as context and it will fails on
Odoo loading

yvaucher committed:
Bump version

yvaucher committed:
Adapt test to mandatory partial attribute in _setup_fields

yvaucher committed:
Adapt tests

yvaucher committed:
Bump version

yvaucher committed:
Address the right container when drawing the widget map

yvaucher committed:
Fix widget initialized too early in second tab thus having bad size

yvaucher committed:
Fix offset issues on the map by reseting offset when scroll bar is used

yvaucher committed:
Don't zoom when setting or changing a geometry

yvaucher committed:
Add an option to set a default zoom to avoid showing a map that you always need to zoom out to get useful informations

yvaucher committed:
No key argument in js, set an optional argument instead with default value

yvaucher committed:
Add an action on map info box to open the selected object in form view

yvaucher committed:
Update Roadmap

yvaucher committed:
Bump version

yvaucher committed:
Fix z-index of Openlayers map container to ensure it doesn't hide search drop down list

yvaucher committed:
Fix unitialized map on 2nd tab of popup forms

yvaucher committed:
Bump version

yvaucher committed:
Bump version

yvaucher committed:
Bump version

yvaucher committed:
Fix geoengine tests with addition of default_zoom

yvaucher committed:
Add selection controlers

yvaucher committed:
Change display of map box info if multiple features are selected

yvaucher committed:
Add button to filter by selected geo features

yvaucher committed:
Implement button to switch in box selection

yvaucher committed:
Improve button to filter features

alan196 committed:
[FIX] from_latlon method srdi was incorrect

damdam-s committed:
bump OL version to 3.18.2

yvaucher committed:
Remove disabling stopdown as it is included in OL3

lmignon committed:
Add click on feature event

yvaucher committed:
Fix Mapinfo box + set cursor pointer

yvaucher committed:
fitExtent is deprecated by fit

yvaucher committed:
Fix css missing bracket

yvaucher committed:
Fix feature counter on view map

yvaucher committed:
Port creation of filter on selection

yvaucher committed:
Improve filter selection button to fix with open single element button

yvaucher committed:
Fix extent error when no feature are present

yvaucher committed:
Set fa icon for geo selection filter

Commit:
Remove MapQuest

see https://github.com/openlayers/ol3/releases/tag/v3.17.0

Commit:
Fix Represenation Mode colored label

yvaucher committed:
Add a selection dragbox when holding SHIFT

yvaucher committed:
[IMP] multi layer selection of records on Geoengine map
count selected records instead of features

Commit:
Fix map size error when open record in a fullscreen map.

damdam-s committed:
geoengine widget first part:
Display the map and geometry plus start point edition

yvaucher committed:
Remove ol alias

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

yvaucher committed:
Define edit layer of geoengine widget

yvaucher committed:
Add modify interaction on geoengine widget (modify interaction needs to be created after map rendering)

yvaucher committed:
Add comment on restrictedExtent

yvaucher committed:
remove comment

yvaucher committed:
Remove google raster layer deprecated option

yvaucher committed:
Remove former ol2 and unused libraries

yvaucher committed:
Remove former fix for scrolling as it is not needed with OL3

yvaucher committed:
Fix onchange for modify control

yvaucher committed:
Fix geostat color legend position

yvaucher committed:
Reimplement button to clear geometry from edition widget

yvaucher committed:
Update license headers

yvaucher committed:
[PORT] base_geoengine to 10.0 - remove old API

yvaucher committed:
jslint

Commit:
[FIX] #132: geo columns were created by odoo base models

mark fields type as manual let geogine the authority to create geom fields

benwillig committed:
[FIX] Column should be created if GeoField is computed and stored

benwillig committed:
[FIX] Cache value for a geo field should be the the hex code. convert_to_record must always return a BaseGeometry instance

benwillig committed:
[ADD] New method to convert 2 points to a LineString

benwillig committed:
[FIX] Fixed tests. Added new tests for new methods

benwillig committed:
[FIX] Pylint

benwillig committed:
[FIX] Handle convert_to_record case when there is no value or if it's not the expected one

benwillig committed:
[FIX] Do not hardcode string during line creation from points

benwillig committed:
[IMP] Allow to define layers which will be active on startup

benwillig committed:
[ADD] Allow to display a configurable popup on map layers.

benwillig committed:
[IMP] Allow to change the size of the drawn circle

benwillig committed:
[IMP] Allow to pass a specific srid to GeoLine.from_points method

dek-odoo committed:
[Add] Allow to display labels on polygons for layers

ecino committed:
[FIX] Replace openerp imports

dek-odoo committed:
Fix Lint & Travis CI issues

dek-odoo committed:
[10, Add] Allow to set opacity on layers (#165)

* Add: Layer opacity

ecino committed:
[FIX] typo in base_geoengine javascript. (#166)

Tonow-c2c committed:
Play python 2to3

Tonow-c2c committed:
Remove coding: utf-8

Tonow-c2c committed:
Readme split in readme folder

yvaucher committed:
Fix signature change on _setup_fields

yvaucher committed:
_select_column_data is replaced by tools.table_columns

commit odoo/d024c7602146ba293c0082303fd476c8613daedc

yvaucher committed:
Change signature of convert_to_column to to introduction of multi values used by Monetary field

yvaucher committed:
Use new MVC approach

asaunier committed:
Fix geoengine view for Odoo 11

asaunier committed:
Fix geoengine widget for Odoo 11

asaunier committed:
Reorganise geoengine files

asaunier committed:
Add base_geoengine README back

Add note about geometry fields to be listed in geoengine view definitions

asaunier committed:
Add support of onchange requests to convert_to_record

asaunier committed:
Update WMTS backgrounds creation

Fix some linting issues

asaunier committed:
Fix JS lint errors

asaunier committed:
Fix travis tests

asaunier committed:
Fix layer switcher

asaunier committed:
Improve handling of maps in tabs

asaunier committed:
Improve WMTS background layers creation

asaunier committed:
Upgrade to OL 3.20.1

lmignon commented on base_geoengine/geo_model.py:
```diff
@@ -0,0 +1,192 @@
+                  '1578017.6490538, 6186191.1800898')
+
+
+class GeoModel(models.BaseModel):
```
Do we still need this specialized model? Why not inheriting from `base`?

yvaucher replied:
I think the same it was not a priority but I agree we can improve this. I'll add it in the todo list.

(a mixin is definitely the way to go)

nikul-serpentcs commented:
@asaunier If it's WIP then change title like `[11.0] [WIP] base geoengine `

yvaucher reviewed (changes requested):
There is a cursor error to fix in travis tests:

```
  File "/home/travis/OCB-11.0/odoo/sql_db.py", line 154, in wrapper
    raise psycopg2.OperationalError(msg)
psycopg2.OperationalError: Unable to use a closed cursor.
```

asaunier committed:
Move base_geoengine tests to a dedicated module

It prevents closed cursor errors, most likely dued to using a mock in the tests.

asaunier commented:
@yvaucher Hopefully 56a359b should fix the closed cursor error reported by Travis.

asaunier commented:
Linting errors have been fixed but Travis still reports an error with sphinxodoo:
> File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/sphinxodoo/ext/autodoc.py", line 43, in load_modules
    import openerp
ImportError: No module named 'openerp'

It seems Travis is stuck with an old version of sphinxodoo (perhaps https://github.com/OCA/odoo-sphinx-autodoc/blob/8a3c37034349a2440f0b96fc6cc50ca877233d50/sphinxodoo/ext/autodoc.py#L43).

@yvaucher How could we make it use the last version that should handle both odoo/openerp?
See https://github.com/OCA/odoo-sphinx-autodoc/blob/master/sphinxodoo/ext/autodoc.py#L48-L52

By the way, could someone remove the "work in progress" label? @max3903 ?

yvaucher reviewed (approved):
LGTM for a first working version

Tests are green only the sphinx doc generation which is broken since ages makes the CI status red.

asaunier commented:
@yvaucher Thanks for reviewing! I have squashed the last 2 fixup commits into 8fd3b3c

Please note this [additional PR](https://github.com/asaunier/geospatial/pull/1) by @smangukiya on top of the current PR. Perhaps to be added as a usual PR when this one is merged?

nikul-serpentcs commented on base_geoengine/doc/build/_static/default.css:
```diff
@@ -0,0 +1,256 @@
+    background-color: #f4debf;
+    border-top: 1px solid #ac9;
+    border-bottom: 1px solid #ac9;
+}
```
``` suggestion
}

```

yvaucher replied:
Please don't use the suggested changes. It is missing EOL we don't want an extra line.

nikul-serpentcs replied:
> Please don't use the suggested changes. It is missing EOL we don't want an extra line.

@yvaucher Here If you did not put a new blank line then Git Error like this,

![test2222222222](https://user-images.githubusercontent.com/17664321/51376918-888f0d00-1b2f-11e9-9d08-1bda91605d54.png)


So better to put EOL

nikul-serpentcs commented on base_geoengine/doc/build/genindex.html:
```diff
@@ -0,0 +1,337 @@
+      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.2.
+    </div>
+  </body>
+</html>
```
```suggestion
</html>

```

nikul-serpentcs commented on base_geoengine/security/data.xml:
```diff
@@ -0,0 +1,14 @@
+        <field name="users" eval="[(4, ref('base.user_root'))]"/>
+    </record>
+
+</odoo>
```
```suggestion
</odoo>

```

max3903 commented:
@asaunier Is it ready to be reviewed and merged or is it still a work in progress to complete all the other checkboxes (like update OL version)?

asaunier commented:
@max3903 It's ready to be reviewed, there is currently no work in progress about this PR. Not sure the remaining checkboxes will be addressed in this PR.

max3903 commented:
@nikul-serpentcs I took care of your feedbacks in #183.
